card/mod/standard/set/all/rich_html/editor.rb

```diff
@@ -0,0 +1,50 @@
+include_set Abstract::ProsemirrorEditor
+include_set Abstract::TinymceEditor
+include_set Abstract::AceEditor
+
+Self::InputOptions.add_to_basket :options, "text area"
+Self::InputOptions.add_to_basket :options, "text field"
+Self::InputOptions.add_to_basket :options, "calendar"
+
+format :html do
+  def editor
+    (c = card.rule(:input)) && c.gsub(/[\[\]]/, "").tr(" ", "_")
+  end
+
+  def editor_method editor_type
+    "#{editor_type}_input"
+  end
+
+  def editor_defined_by_card
+    return unless (editor_card = Card[editor])
+    nest editor_card, view: :core
+  end
+
+  view :editor do
+    try(editor_method(editor)) ||
+      editor_defined_by_card ||
+      send(editor_method(default_editor))
+  end
+
+  def default_editor
+    :rich_text
+  end
+
+  # overriden by mods that provide rich text editors
+  def rich_text_input
+    prosemirror_editor_input
+  end
```

Comment:
The set_pattern change was necessary because self sets didn't work for junction cards. The issue was the following: 
Self sets have only one anchor according to their set pattern definition, so when the sets are loaded a set in `/self/input/right/options` is considered to be a self set for the card `input` and the set module is stored under the key `Self::Input`. But the key that is generated for `*input+*right+*options` to lookup the set modules is always `Self::Input::Right::Options` even if `*input+*right+*options` has its own codename.
I changed it so that if a junction card has a codename then that codename is used instead of the composite of the codenames of its parts.